server/src/internalClusterTest/java/org/elasticsearch/index/mapper/DynamicMappingIT.java

```diff
@@ -156,4 +171,119 @@ public void testMappingVersionAfterDynamicMappingUpdate() throws Exception {
         client().prepareIndex("test").setId("1").setSource("field", "text").get();
         assertBusy(() -> assertThat(clusterService.state().metadata().index("test").getMappingVersion(), equalTo(1 + previousVersion)));
     }
+
+    public void testBulkRequestWithDynamicTemplates() throws Exception {
+        final XContentBuilder mappings = XContentFactory.jsonBuilder();
+        mappings.startObject();
+        {
+            mappings.startArray("dynamic_templates");
+            {
+                mappings.startObject();
+                mappings.startObject("location");
+                {
+                    if (randomBoolean()) {
+                        mappings.field("match", "location");
+                    }
+                    if (randomBoolean()) {
+                        mappings.field("match_mapping_type", "string");
+                    }
```

Comment:
I think I would try to avoid randomizing behaviour in an IT test. They tend to be hard to reproduce. If this is to have more coverage, shall we have one test method per scenario that we want to test?

Reply:
> They tend to be hard to reproduce.

We have a lot of improvements in IT tests recently, now they are almost reproducible as unit tests. Are you okay if I leave this test as is?

Reply:
ok